cli/tauri-cli/src/bundle/wix.rs

```diff
@@ -60,6 +60,52 @@ struct ExternalBinary {
   path: String,
 }
 
+#[derive(Serialize)]
+struct ResourceDirectory {
+  name: String,
+  files: Vec<ResourceFile>,
+  directories: Vec<ResourceDirectory>
+}
+
+impl ResourceDirectory {
+  fn add_file(&mut self, file: ResourceFile) {
+    self.files.push(file);
+  }
+
+  // generates the wix XML string to bundle this directory resources recursively
+  fn get_wix_data(self) -> (String, Vec<String>) {
+    let mut files = String::from("");
+    let mut file_ids = Vec::new();
+    for file in self.files {
+      file_ids.push(file.id.clone());
+      files.push_str(
+        format!(
+          r#"<Component Id="{id}" Guid="{guid}" Win64="$(var.Win64)" KeyPath="yes"><File Id="PathFile_{id}" Source="{path}" /></Component>"#,
+          id = file.id,
+          guid = file.guid,
+          path = file.path
+        ).as_str()
+      );
+    }
+    let mut directories = String::from("");
+    for directory in self.directories {
+      let (wix_string, ids) = directory.get_wix_data();
+      for id in ids {
+        file_ids.push(id)
+      }
+      directories.push_str(wix_string.as_str());
+    }
+    (format!(r#"<Directory Id="{name}" Name="{name}">{contents}</Directory>"#, name = self.name, contents = format!("{}{}", files, directories)), file_ids)
```

Comment:
Refactor the large `format!` string into a let binding to make the code a bit clearer.